addon/instance-initializers/ember-custom-elements.js

```diff
@@ -63,3 +57,26 @@ export function initialize(instance) {
 export default {
   initialize
 };
+
+const DECORATOR_REGEX = /customElement\s*\){0,1}\s*\(/;
+
+function determineIfShouldEvalModule(instance, _module) {
+  const {
+    emberCustomElements = {}
+  } = instance.resolveRegistration('config:environment');
+  if (emberCustomElements.deoptimizeModuleEval) return true;
+  function _moduleShouldEval(_module) {
+    for (const moduleName of _module.deps) {
+      // Check if ember-custom-elements is a dependency of the module
+      if (moduleName === 'ember-custom-elements') {
+        const code = (_module.callback || function() {}).toString();
+        // Test if a function named "customElement" is called within the module
+        if (DECORATOR_REGEX.test(code)) return true;
+      }
+      const dep = instance.__registry__.fallback.resolver._moduleRegistry._entries[moduleName];
+      if (dep && _moduleShouldEval(dep)) return true;
+    }
+    return false;
+  }
+  return _moduleShouldEval(_module);
```

Comment:
Why define this function and then immediately call it, rather than just invoking the function contents here directly?

It seems like this could instead be

```suggestion
  for (const moduleName of _module.deps) {
    // Check if ember-custom-elements is a dependency of the module
    if (moduleName === 'ember-custom-elements') {
      const code = (_module.callback || function() {}).toString();
      // Test if a function named "customElement" is called within the module
      if (DECORATOR_REGEX.test(code)) return true;
    }
    const dep = instance.__registry__.fallback.resolver._moduleRegistry._entries[moduleName];
    if (dep && _moduleShouldEval(dep)) return true;
  }
  return false;
```

Reply:
It's so that we don't have to resolve the app config at every level of recursion.  No other reason.